### PR TITLE
Add APIs for get the current routes configuration

### PR DESCRIPTION
**Pull request by Amygos**
This PR will add two new APIs
- get-route`:  get the current configuration of a specific route
- 'list-routes': get a list of the currently configured routes

It Is also provided a new test suite for all the current get/set/list/delete routes APIs
See the updated README for the full documentation.

### CODE REVIEW

File: tests/api.resource [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+*** Keywords ***
+Run task
+    [Arguments]    ${action}    ${input}    ${decode_json}=${TRUE}    ${rc_expected}=0
+    ${stdout}    ${stderr}    ${rc} =     Execute Command    api-cli run ${action} --data '${input}'    return_stdout=True    return_stderr=True    return_rc=True
+    Should Be Equal As Integers    ${rc_expected}    ${rc}    Run task ${action} failed!${\n}${stderr}
+    IF    ${decode_json}
+        ${response} =    Evaluate    json.loads("""${stdout}""")    modules=json
+    ELSE
+        ${response} =    Set Variable    ${stdout}
+    END
```
Comment (DavidePrincipi) [inline tests/api.resource:13]: Please see my recent change here:

https://github.com/NethServer/ns8-core/pull/235/files#diff-07e675b43ff00a8905a236625dd1d023dc28a6af9d793aec4b921f1b448a322c

What do you think?

Comment (Amygos) [reply tests/api.resource:13]: Seem fine to me, I will try the PR with that changes.

**Review by gsanchietti — APPROVED** [review]

File: imageroot/actions/get-route/20readconfig [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+# Try to parse the stdin as JSON.
+# If parsing fails, output everything to stderr
+data = json.load(sys.stdin)
+module = data['instance']
+route = {}
+
+api_path = os.getenv("API_PATH", "")
+if not api_path:
+    raise Exception("API_PATH not found inside the environemnt")
```
Comment (DavidePrincipi) [inline imageroot/actions/get-route/20readconfig:38]: If API_PATH is missing it's ok to not handle the KeyError and exit immediately with the native exception

```suggestion
api_path = os.environ["API_PATH"]
```

File: imageroot/actions/list-routes/20readconfig [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import json
+import os
+import agent
+import sys
+import urllib.request
+
+
+api_path = os.getenv("API_PATH", "")
+if not api_path:
+    raise Exception("API_PATH not found inside the environemnt")
```
Comment (DavidePrincipi) [inline imageroot/actions/list-routes/20readconfig:32]:
```suggestion
api_path = os.environ["API_PATH"]
```

...same as above

**Review by DavidePrincipi — APPROVED** [review]
It's ok, I want to propose just a couple of Python code cleanups

**Pull request merged**